models/movies/search.js

```diff
@@ -0,0 +1,42 @@
+
+// Handle Movies Search
+
+var pageglobal = 1;
+var totalPages = 1;
+
+function requestApi(page) {
+
+    var url = "https://api.themoviedb.org/3/search/movie?api_key=" + key + "&language=en-US&query=" + searchQuery + "&page=" + page + "&include_adult=false";
+
+    if (totalPages > page) {
+        document.getElementById('loadmore').style.display = 'none';
+        return false;
+    }
+
+    AJAX.request(url,function() {
+        if (!result.status_code) {
+            document.getElementById('loadmore').style.display = 'inline';
+            document.getElementById('loading').style.display = 'none';
+            result.results.forEach(renderCards);
+
+        } else {
+            console.log(result.status_code);
+        }
+    });
+    pageglobal++;
+}
+
+function renderCards(movie, i) {
+    if ('content' in document.createElement('template')) {
+        var c = document.querySelector('.single_card').content;
+
+        c.querySelector('h3').textContent = movie.original_title;
+        c.querySelector('img').src = 'https://image.tmdb.org/t/p/w300/' + movie.poster_path;
+        c.querySelector('.btn').href = "?controller=movies&action=detailed&movie_id=" + movie.id;
+
+        c.querySelector('.release').textContent = movie.release_date;
+        document.querySelector('#render').appendChild(document.importNode(c, true));
+    }
+}
+
+requestApi(pageglobal);
```

Comment:
It seems to me that this call should be encapsulated to be called when needed. Looks like it'll be called when ever you include this file in a page. What do you think?

Reply:
Yea. It would be much better if was created a class with the 2 methods and encapsulated the call..